corehq/apps/user_importer/helpers.py

```diff
@@ -252,6 +252,25 @@ def update_locations(self, location_codes, domain_info):
             else:
                 self.logger.add_info(UserChangeMessage.primary_location_removed())
 
+    def update_user_groups(self, domain_info, group_names):
+        try:
+            old_group_ids = set()
+            for group in domain_info.group_memoizer.by_user_id(self.user.user_id):
+                old_group_ids.add(group.get_id)
+                if group.name not in group_names:
+                    group.remove_user(self.user)
+
+            new_groups = {}
+            for group_name in group_names:
+                group = domain_info.group_memoizer.by_name(group_name)
+                group.add_user(self.user, save=False)
+                new_groups[group.get_id] = group
+
+            if set(new_groups) != old_group_ids:
+                self.logger.add_info(UserChangeMessage.groups_info(list(new_groups.values())))
+        except Exception as e:
```

Comment:
Can you share why we have a generic exception catch here?

Reply:
The generic exception is my Node/JavaScript instinct. It's more Pythonic to anticipate specific exceptions and I'll get into the habit of doing it that way. Perhaps we don't need to handle a possible error here since `update_user_groups` is called inside the try block in `create_or_update_commcare_users_and_groups`. What do you think?

Reply:
Not sure about the comparison between languages, but generally I think Generic exceptions are needed in very rare scenarios. 
And ya, good point about the parent try block. In this block of code if we don't expect an exception, we should not have an except (`except when expect` if you would say :) )